DigitalLearningSolutions.Data/Services/TutorialContentService.cs

```diff
@@ -116,8 +125,33 @@ FROM Tutorials
                               ELSE Sections.PLAssessPath
                          END AS PostLearningAssessmentPath,
                          Applications.CourseSettings,
-                         NextTutorial.TutorialID AS NextTutorialID,
-                         NextSection.SectionID AS NextSectionID
+                         Applications.IncludeCertification,
+                         Customisations.IsAssessed,
+                         Progress.Completed,
+                         Applications.AssessAttempts AS MaxPostLearningAssessmentAttempts,
+                         Applications.PLAPassThreshold AS PostLearningAssessmentPassThreshold,
+                         Customisations.DiagCompletionThreshold AS DiagnosticAssessmentCompletionThreshold,
+                         Customisations.TutCompletionThreshold AS TutorialsCompletionThreshold,
+                         NextTutorial.NextTutorialID,
+                         NextSection.NextSectionID,
+                         CAST (CASE WHEN EXISTS(SELECT 1
+                                                  FROM OtherTutorials
+                                                 WHERE SectionID <> @sectionId)
+                                         THEN 1
+                                    ELSE 0
+                               END AS BIT) AS OtherSectionsExist,
+                         CAST (CASE WHEN (CustomisationTutorials.DiagStatus = 1 AND Sections.DiagAssessPath IS NOT NULL)
+                                         OR (Customisations.IsAssessed = 1 AND Sections.PLAssessPath IS NOT NULL)
+                                         OR Sections.ConsolidationPath IS NOT NULL
+                                         THEN 1
+                                    WHEN EXISTS(SELECT 1
+                                                  FROM OtherTutorials
+                                                 WHERE SectionID = @sectionId
+                                                       AND (Status = 1 OR HasDiagnostic = 1)

```

Comment:
Why do we need to include HasDiagnostic here? Should that be included in the `WHEN` above? 🙂

Reply:
So the `WHEN` above checks whether this tutorial is part of the diagnostic assessment, whereas this bit checks whether the other tutorials in the section are part of the diagnostic assessment. If either are true then there is a diagnostic assessment. The reason there are two checks is because the current tutorial is not part of the `OtherTutorials` 🙂

Reply:
Ah I see! 🙂